equinix/resource_fabric_cloud_router.go

```diff
@@ -273,6 +313,10 @@ func resourceFabricCloudRouterCreate(ctx context.Context, d *schema.ResourceData
 	package_ := packageCloudRouterTerraformToGo(schemaPackage)
 	createCloudRouterRequest.SetPackage(package_)
 
+	schemaMarketplaceSubscription := d.Get("marketplace_subscription").(*schema.Set).List()
+	marketplaceSubscription := marketplaceSubscriptionCloudRouterTerraformToGo(schemaMarketplaceSubscription)
+	createCloudRouterRequest.SetMarketplaceSubscription(marketplaceSubscription)
```

Comment:
Should only set the subscription if the value is provided. Do a reflect deep equal check on empty return struct to avoid empty value in JSON request body.